tests/integration/ha/test_scaling.py

```diff
@@ -0,0 +1,153 @@
+
+
+async def recreate_opensearch_kibanaserver(ops_test: OpsTest):
+    """Temporary helper function."""
+    #
+    # THIS HAS TO CHANGE AS https://warthogs.atlassian.net/browse/DPE-2944 is processed
+    #
+    # "Total Hack"
+    # Currently the 'kibanaserver' user is deleted on opensearch
+    # We are "re-adding" it so we could use it for the opensearch connection
+    # We are "re-adding" it using the password shared on the relation for the opensearch-client_<id> user
+    # that's currently used by the charm
+    #
+    # To make it EVEN worse: we set the opensearch charm update period to 1h,
+    # since on each status update opensearch is deleting all "unexpected" users :sweat_smile:
+    #
+    opensearch_endpoints = await get_application_relation_data(
+        ops_test, APP_NAME, "opensearch_client", "endpoints"
+    )
+    opensearch_endpoint = opensearch_endpoints.split(",")[0]
+
+    unit_name = f"{OPENSEARCH_CHARM}/0"
+    action = await ops_test.model.units.get(unit_name).run_action("get-password")
+    await action.wait()
+    opensearch_admin_password = action.results.get("password")
+
+    dashboard_credentials = await get_secret_by_label(
+        ops_test, f"opensearch-client.{pytest.relation.id}.user.secret"
+    )
+    dashboard_password = dashboard_credentials["password"]
+    set_opensearch_user_password(
+        opensearch_endpoint, opensearch_admin_password, dashboard_password
+    )
+
+    await ops_test.model.wait_for_idle(
+        apps=[OPENSEARCH_CHARM, APP_NAME], status="active", timeout=1000
+    )
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+@pytest.mark.charm
+async def test_deploy_active(ops_test: OpsTest):
+
+    charm = await ops_test.build_charm(".")
+    await ops_test.model.deploy(charm, application_name=APP_NAME, num_units=1)
+    await ops_test.model.set_config(OPENSEARCH_CONFIG)
+    # Pinning down opensearch revision to the last 2.10 one
+    # NOTE: can't access 2/stable from the tests, only 'edge' available
+    await ops_test.model.deploy(OPENSEARCH_CHARM, channel="2/edge", num_units=1)
+
+    config = {"ca-common-name": "CN_CA"}
+    await ops_test.model.deploy(TLS_CERTIFICATES_APP_NAME, channel="stable", config=config)
+
+    await ops_test.model.wait_for_idle(
+        apps=[TLS_CERTIFICATES_APP_NAME], status="active", timeout=1000
+    )
+
+    # Relate it to OpenSearch to set up TLS.
+    await ops_test.model.relate(OPENSEARCH_CHARM, TLS_CERTIFICATES_APP_NAME)
+    await ops_test.model.wait_for_idle(
+        apps=[OPENSEARCH_CHARM, TLS_CERTIFICATES_APP_NAME], status="active", timeout=1000
+    )
+
+    async with ops_test.fast_forward():
+        await ops_test.model.block_until(
+            lambda: len(ops_test.model.applications[APP_NAME].units) == 1
+        )
+        await ops_test.model.wait_for_idle(
+            apps=[APP_NAME], status="active", timeout=1000, idle_period=30
+        )
+
+    assert ops_test.model.applications[APP_NAME].status == "active"
```

Comment:
That decision is normally outside of the scope of this task.

But it's a valid question. Do we want to do that?
(It would probably go in a separate PR as multiple tests would be impacted.)

Reply:
This issue was supposed to be highlighted on the Charm integration tests. It's not the scaling tests that "show" it.
We just didn't notice it earlier.

A status change is not only unrelated to Scaling tests, but requires its own tests, etc... I prefer to keep tasks self-contained and... not to endlessly extend this one in case we may come across issues (like a deadlock ;-) ) when adding new content.

https://warthogs.atlassian.net/browse/DPE-4167 is added

Reply:
Let's add the status to this PR please. We are deferring too many tasks to the future - and I'm worried that none will get addressed in due time. The status check only adds 1 line to the tests.

Reply:
I'm failing to follow why I may have more capacity to do a task because it is in a separate ticket.

The time spent on the task is equal.

Reply:
Sorry, let's do (absolutely necessary) status changes as a separate task.